components/scream/src/control/atmosphere_driver.cpp

```diff
@@ -139,21 +139,28 @@ void AtmosphereDriver::initialize (const ekat::Comm& atm_comm,
   //         phys-dyn remapper (which uses homme's mpi infrastructure).
   m_atm_process_group->setup_remappers(*m_field_repo);
 
-  // Initialize atm inputs
-  init_atm_inputs ();
-
-  // Now we can inspect the dag, including also checking that
-  // all fields are correctly inited. Any unmet dependency in
-  // the dag at this point has to be treated as an error.
-  inspect_atm_dag ();
-
   // Set time steamp t0 to all fields
   for (auto& field_map_it : *m_field_repo) {
     for (auto& f_it : field_map_it.second) {
       f_it.second.get_header().get_tracking().update_time_stamp(t0);
     }
   }
 
+#if defined(SCREAM_CIME_BUILD)
+  // If this is a CIME build, we need to prepare the surface coupling
+  m_surface_coupling = std::make_shared<SurfaceCoupling>(m_grids_manager->get_reference_grid(),*m_field_repo);
+#endif
+
+  if (!m_surface_coupling) {
+    // Standalone runs *may* require initialization of atm inputs.
+    init_atm_inputs ();
+
+    // In standalone runs, we already have everything we need for the dag,
+    // so we can proceed to do its inspection. Any unmet dependency in
+    // the dag at this point has to be treated as an error.
+    inspect_atm_dag ();
```

Comment:
It's not like that. The DAG cannot be inspected _at this moment_, since SC is not yet setup. If you look in `mct_coupling/scream_cxx_f90_interface.cpp`, in `scream_setup_surface_coupling(...)`, you will see that we _do_ inspect the DAG, but only when everything has been completely inited. For standalone runs, it can be done inside ad.initialize, since we don't have any other source for the fields, but for CIME runs it has to wait.